.github/workflows/test.yml

```diff
@@ -26,16 +26,59 @@ jobs:
         shell: bash
         run: make lint
 
+  prepare-unit-test-matrix:
```

Comment:
Prepares matrix's based on push/pull request vs workflow dispatch. Note: matrix's are the same for unit tests, but defined separately in case changes need to be made in the future.